test/testtokenize.cpp

```diff
@@ -6938,6 +6938,16 @@ class TestTokenizer : public TestFixture {
                             InternalError,
                             "There is an unknown macro here somewhere. Configuration is required. If MACRO is a macro then please configure it.");
 
+        ASSERT_THROW_EQUALS(tokenizeAndStringify("void f(int i) {\n" // #11770
+                                                 "    if (i == 0) {}\n"
+                                                 "    else if (i == 1) {}\n"
+                                                 "    else\n"
+                                                 "        MACRO(i)\n"
```

Comment:
I assume that if there is a `;` after the `MACRO(i)` then no unknownMacro will be written?